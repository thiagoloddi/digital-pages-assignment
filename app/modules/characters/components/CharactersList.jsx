import React from 'react';
import injectSheet from 'react-jss';
import strftime from 'strftime';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import CharacterThumb from './CharacterThumb';

const time = strftime.localizeByIdentifier('pt_BR');

const styles = {
  table: {
    '& td, & th': {
      fontSize: '14px'
    }
  },
  name: {
    width: '300px'
  },
  modified: {
    width: '200px'
  },
  row: {
    '&:hover': {
      backgroundColor: '#f5f5f5',
      cursor: 'pointer'
    }
  }
}

const CharactersList = ({ characters, classes, onRowClick }) => {
  const renderBody = () => {
    return characters.map(({ id, name, description, modified, thumbnail: { path, extension } }) => (
      <TableRow onClick={onRowClick} id={id} className={classes.row} key={id}>
        <TableCell className={classes.name}><CharacterThumb name={name} src={`${path}.${extension}`}/></TableCell>
        <TableCell>{description || "-"}</TableCell>
        <TableCell align="center">{modified ? time("%d/%m/%y - %H:%M", new Date(modified)) : "-"}</TableCell>
      </TableRow>
    ));
  }

  return (
    <div>
      <Paper>
        <Table className={classes.table}>
          <TableHead>
            <TableRow>
              <TableCell>Nome</TableCell>
              <TableCell >Descrição</TableCell>
              <TableCell className={classes.modified} align="center">Última Atualização</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {renderBody()}
          </TableBody>
        </Table>
      </Paper>
    </div>
  );
}

export default injectSheet(styles)(CharactersList);