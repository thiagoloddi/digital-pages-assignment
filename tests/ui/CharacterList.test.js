import React from 'react';
import renderer from 'react-test-renderer';
import CharactersList from '../../app/modules/characters/components/CharactersList';

const createNodeMock = e => {
  if(e.type == 'img') {
    return {};
  }
}

const func = () => {};

const characters = [
  { id: "1", name: "Char1", description: "Description 1", modified: 1, thumbnail: { path: "path", extension: ".jpg" }},
  { id: "2", name: "Char1", description: "Description 2", modified: 2, thumbnail: { path: "path", extension: ".jpg" }}
];

it('renders correctly', () => {
  const tree = renderer
    .create(<CharactersList characters={characters} onRowClick={func} />, { createNodeMock })
    .toJSON();
  expect(tree).toMatchSnapshot();
});