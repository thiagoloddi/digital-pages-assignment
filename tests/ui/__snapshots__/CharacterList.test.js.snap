// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
  >
    <table
      className="MuiTable-root-28 CharactersList-table-0-1-1"
    >
      <thead
        className="MuiTableHead-root-29"
      >
        <tr
          className="MuiTableRow-root-30 MuiTableRow-head-33"
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-head-36"
            scope="col"
          >
            Nome
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-head-36"
            scope="col"
          >
            Descrição
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-head-36 MuiTableCell-alignCenter-44 CharactersList-modified-0-1-3"
            scope="col"
          >
            Última Atualização
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root-47"
      >
        <tr
          className="MuiTableRow-root-30 CharactersList-row-0-1-4"
          id="1"
          onClick={[Function]}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37 CharactersList-name-0-1-2"
          >
            <div
              className="CharacterThumb-container-0-1-5"
            >
              <div
                className="Image-container-0-1-7 "
              >
                <img
                  className="Image-img-0-1-8"
                  src="path..jpg"
                  style={
                    Object {
                      "visibility": "hidden ",
                    }
                  }
                />
                <img
                  className="Image-loading-0-1-9"
                  src="/assets/progress.gif"
                  style={null}
                />
              </div>
              <div
                className="CharacterThumb-name-0-1-6"
              >
                Char1
              </div>
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37"
          >
            Description 1
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignCenter-44"
          >
            31/12/69 - 21:00
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-30 CharactersList-row-0-1-4"
          id="2"
          onClick={[Function]}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37 CharactersList-name-0-1-2"
          >
            <div
              className="CharacterThumb-container-0-1-5"
            >
              <div
                className="Image-container-0-1-7 "
              >
                <img
                  className="Image-img-0-1-8"
                  src="path..jpg"
                  style={
                    Object {
                      "visibility": "hidden ",
                    }
                  }
                />
                <img
                  className="Image-loading-0-1-9"
                  src="/assets/progress.gif"
                  style={null}
                />
              </div>
              <div
                className="CharacterThumb-name-0-1-6"
              >
                Char1
              </div>
            </div>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37"
          >
            Description 2
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignCenter-44"
          >
            31/12/69 - 21:00
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
